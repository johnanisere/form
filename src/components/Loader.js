import React from "react";
import { BeatLoader } from "react-spinners";
import ActivityIndicator from "../blocks/ActivityIndicator";

const Loader = () => (
  <ActivityIndicator>
    <BeatLoader size={9} />
  </ActivityIndicator>
);

export default Loader;
