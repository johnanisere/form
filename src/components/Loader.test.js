import React from "react";
import { shallow } from "enzyme";

import Loader from "./Loader";

it("Loader component exists", () => {
  const wrapper = shallow(<Loader />);
  expect(wrapper.exists()).toBe(true);
});

it("Loader component renders", () => {
  const wrapper = shallow(<Loader />);
  expect(wrapper).toMatchSnapshot();
});
