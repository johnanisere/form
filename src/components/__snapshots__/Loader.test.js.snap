// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Loader />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <onlyUpdateForKeys(Loader)
        color="#000000"
        css={Object {}}
        loading={true}
        margin="2px"
        size={9}
        sizeUnit="px"
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "color": "#000000",
        "css": Object {},
        "loading": true,
        "margin": "2px",
        "size": 9,
        "sizeUnit": "px",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bwzfXH",
        "isStatic": true,
        "rules": Array [
          "
  justify-content: center;
  align-items: center;
  height: 100%;
  width: 100%;
  display: flex;
",
        ],
      },
      "displayName": "Styled(styled.aside)",
      "foldedComponentIds": Array [
        "sc-bdVaJa",
      ],
      "render": [Function],
      "styledComponentId": "sc-bwzfXH",
      "target": "aside",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <onlyUpdateForKeys(Loader)
          color="#000000"
          css={Object {}}
          loading={true}
          margin="2px"
          size={9}
          sizeUnit="px"
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "color": "#000000",
          "css": Object {},
          "loading": true,
          "margin": "2px",
          "size": 9,
          "sizeUnit": "px",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": true,
          "rules": Array [
            "
  justify-content: center;
  align-items: center;
  height: 100%;
  width: 100%;
  display: flex;
",
          ],
        },
        "displayName": "Styled(styled.aside)",
        "foldedComponentIds": Array [
          "sc-bdVaJa",
        ],
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": "aside",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
